src/main/java/core/basesyntax/service/strategy/PurchaseFruitTransactionHandler.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseFruitTransactionHandler implements FruitTransactionHandler {
+    private final Storage storage;
+
+    public PurchaseFruitTransactionHandler(Storage storage) {
+        this.storage = storage;

```

Constructor does not validate its dependency. Per the task guidance you should check null parameters and fail fast with an informative message. Add a null-check for `storage` in the constructor and throw an informative RuntimeException if it's null (for example: `throw new RuntimeException("Storage must not be null")`). This enforces dependency correctness and improves testability.